Displays the frequencies of the chars to the user

TiagoMaiaL opened the pull request:
In the process of analyzing a text, the program now prints the percentage of the chars according to the length of the text. Before this change, the statistics would include the number of occurrences of a char, this new approach is more intuitive.

TiagoMaiaL committed:
Add the getItemFrequencies() definition and its tests

Tests the expected behavior of getItemFrequences, which takes
a total count of chars and the count of each chars as a Map,
and returns the computed frequency Map.

TiagoMaiaL committed:
Implement itemFrequencies()

TiagoMaiaL committed:
Assert that a valid counts map is always passed into itemFrequencies

This avoids unexpected behavior when calling this function.

TiagoMaiaL committed:
Move the tests of 'itemFrequencies' and 'sanitize' out of 'getChars'

TiagoMaiaL committed:
Print the chars frequencies instead of their count

This still needs some good formatting

TiagoMaiaL committed:
Format the float frequency numbers to print percent floats

TiagoMaiaL committed:
Test the function that generates the statistics display text

jfarmer commented on textalyze.js:
```diff
@@ -39,6 +39,31 @@ function itemCounts(array) {
+ */
+function itemFrequencies(totalCount, itemCounts) {
+  if (totalCount <= 0) {
+    throw new Error('The items frequencies can\'t be computed without a total count of chars.');
```
Throw the most specific error that makes sense.  In this case, [RangeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError) is what you want.

That said, you don't really need to pass in `totalCount` since you can calculate it by summing `itemCounts.values()`:

```javascript
const totalCount = Array.from(itemCounts.values()).reduce((x, y) => x + y);
```

TiagoMaiaL replied:
Much more expressive than using anonymous functions. This syntax is very similar to Swift closures.

In this case, I still need to pass the initial value. When the array is empty, it complains of no value being passed as the initial one.

TiagoMaiaL replied:
Thanks for showing me that.

jfarmer commented on textalyze.js:
```diff
@@ -39,6 +39,31 @@ function itemCounts(array) {
+  }
+
+  if (typeof itemCounts != 'object' || !(itemCounts instanceof Map)) {
+    throw new Error('The item frequencies must be a valid map.');
```
Here you'd want [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError).

TiagoMaiaL committed:
Refactor item counts to return map directly after setting a value

Since map.set() alerady returns it, it's not necessary to return
it later.

TiagoMaiaL committed:
Fix sanitized typo in docs

TiagoMaiaL committed:
Compute item frequencies solely form the counts map

There's no need to passing the total count, since it can be
computed from the passed map of char occurrences.

TiagoMaiaL committed:
Refactor error display when no files are passed as arguments

Intead of throwing an error, it's more intuitive to simply
display an error message.

TiagoMaiaL committed:
Refactor closure expression to a more readable sum form

jfarmer commented on textalyze.js:
```diff
@@ -33,9 +33,7 @@ function getChars(text) {
-    map.set(value, currentCount + 1);
-
-    return map;
+    return map.set(value, currentCount + 1);
```
👍
